graphql_compiler/tests/schema_transformation_tests/test_rename_schema.py

```diff
@@ -21,6 +22,38 @@
 from .input_schema_strings import InputSchemaStrings as ISS
 
 
+def validate_error_message_component(
+    error_message_component: str, expected_prefix: str, expected_data_structure: Any
+) -> bool:
+    """Confirm that error_message_component is as expected.
+
+    A common pattern when checking string representations of exceptions here is to have the message
+    consist of a string literal prefix, which describes the error reasoning, followed by the string
+    representation of a data structure that describes which parts of the schema caused the error.
+    This function validates the error message based on the expected prefix and expected data
+    structure. This is useful when the data structure's __str__() method is not deterministic
+    (e.g. set).
+
+    Args:
+        error_message_component: error message part to validate
+        expected_prefix: explanation of why the error was raised
+        expected_data_structure: description of which parts of the schema caused the error
+
+    Returns:
+        True iff the error message correctly consists of the prefix followed by the expected data
+        structure.
+    """
```

Comment:
To be honest, this feels like way overkill and also like testing code by writing more (and more complex) code. Testing error messages by relying on a correct implementation of this function, including uses of `literal_eval` etc., seems problematic.

If non-determinism is the source of the problem, let's tackle it directly — output the items in sorted order in the error message. That will allow you to write out the actual error message strings in the tests, which would make it much easier for reviewers and readers of the code to review the content of the error message and evaluate its usefulness.